docs/src/main/sphinx/functions/setdigest.rst

```diff
@@ -0,0 +1,156 @@
+         ),
+         setdigest_side_by_side(id1, digest1, id2, digest2) AS (
+             SELECT m1.id as id1,
+                    m1.digest as digest1,
+                    m2.id as id2,
+                    m2.digest as digest2
+             FROM (SELECT id, digest FROM minhash_digest) m1
+             JOIN (SELECT id, digest FROM minhash_digest) m2
+               ON m1.id != m2.id AND m1.id < m2.id
+        )
+    SELECT id1,
+           id2,
+           intersection_cardinality(digest1, digest2) AS intersection_cardinality,
+           jaccard_index(digest1, digest2)            AS jaccard_index
+    FROM setdigest_side_by_side
+    ORDER BY id1, id2;
+
+.. code-block:: text
+
+     id1 | id2 | intersection_cardinality |    jaccard_index
+    -----+-----+--------------------------+---------------------
+       1 |   2 |                        3 | 0.16666666666666666
+       1 |   3 |                        0 |                 0.0
+       1 |   4 |                        4 |                 0.6
+       2 |   3 |                        0 |                 0.0
+       2 |   4 |                        1 |                 0.0
+       3 |   4 |                        0 |                 0.0
+
+Serialization
+-------------
+
+Data sketches can be serialized to and deserialized from ``varbinary``. This
+allows them to be stored for later use.
+
+
+Functions
+---------
+
+.. function:: make_set_digest(x) -> setdigest
+
+    Composes all input values of ``x`` into a ``setdigest``.
+
+    ``x`` must be of ``bigint`` type.
+
+
+.. function:: merge_set_digest(setdigest) -> setdigest
+
+    Returns the ``setdigest`` of the aggregate union of the individual ``setdigest``
+    Set Digest structures.
+
+.. function:: cardinality(setdigest) -> long
+    :noindex:
+
+    Returns the cardinality of the set digest from its internal
+    ``HyperLogLog`` component.
+
+    Examples::
+
+        SELECT cardinality(make_set_digest(value))
+        FROM (VALUES 1, 2, 2, 3, 3, 3, 4, 4, 4, 4, 5) T(value);
+        -- 5
+
+.. function:: intersection_cardinality(x,y) -> long
+
+    Returns the estimation for the cardinality of the intersection of the two set digests.
+
+    ``x`` and ``y`` must be of type  ``setdigest``
+
+    Examples::
+
+        SELECT intersection_cardinality(make_set_digest(v1), make_set_digest(v2))
+        FROM (VALUES (1, 1), (NULL, 2), (2, 3), (3, 4)) T(v1, v2);
+        -- 3
+
+.. function:: jaccard_index(x, y) -> double
+
+    Returns the estimation of `Jaccard index <https://en.wikipedia.org/wiki/Jaccard_index>`_ for
+    the two set digests.
+
+    ``x`` and ``y`` must be of type  ``setdigest``.
+
+    Examples::
+
+        SELECT jaccard_index(make_set_digest(v1), make_set_digest(v2))
+        FROM (VALUES (1, 1), (NULL,2), (2, 3), (NULL, 4)) T(v1, v2);
+        -- 0.5
+
+.. function:: hash_counts(x) -> map(bigint, smallint)
+
+    Returns a map containing the `Murmur3Hash128 <https://en.wikipedia.org/wiki/MurmurHash#MurmurHash3>`_
```

Comment:
Maybe also link to the murmurhash function docs?

Reply:
It is true that internally, this routine makes used of the `io.airlift.slice.Murmur3Hash128`.
However, compared to `murmur3` scalar function implementation, the code for the `hash_counts` function makes use of  the method `io.airlift.slice.Murmur3Hash128#hash64(io.airlift.slice.Slice)`

I wouldn't say that the link to the  `murmur3` hash function documentation is appropriate here.

Reply:
fair enough